### DISCUSSION TITLE
Vector's Built-in-Redundancy Mechanism

### DISCUSSION THREAD

**Discussion by hemanthofficial3009** [vectordotdev/vector#11683] [Q&A] [answered]
Hello,
We run vector instances on different machines which acts as Agents and Collector.

_**http end-point -> Agent (source type="http") (sink type ="vector") -> Transform (type="remap") -> Collector (source type="vector") (sink type="file")**_

- Vector acting as an agent would received data from an Http end-point. 
- The data received would be enriched (transform) and passed to a down stream vector object which acts as a collector running on another machine.

Can you let us know if there is an option for retries on the sink of the agent which can help to identify that the collector instance is not responding

**Comment by spencergilbert** [comment, marked as answer]
The `vector` sink will retry requests based on the provided configuration here: https://vector.dev/docs/reference/configuration/sinks/vector/#request

**Reply by hemanthofficial3009** [reply]
Thanks @spencergilbert for the quick response. 
Can you throw some light on the redundancy mechanism provided by vector, in case any of the configured sink fails to receive the events.

**Reply by spencergilbert** [reply]
See:

https://vector.dev/docs/about/under-the-hood/networking/arc/
https://vector.dev/docs/about/under-the-hood/architecture/end-to-end-acknowledgements/
https://vector.dev/docs/about/concepts/#buffers

**Reply by hemanthofficial3009** [reply]
@spencergilbert that was informative, thanks for sharing that.
use case :
each of the below vector instances are running on different systems in a network.
- vector instance acting as an _Agent (X)_ 
- vector instance acting as primary _Collector (Y)_
- vector instance acting as secondary _Collector (Z)_
- Regular case where Y is running : X ------> Y 
- Fail over case : if Y is down then only X ------> Z

X is configured to sink data to Y. for some reason the system where the primary Collector (Y) is present went down.

In this case is there any feature vector provides as part of configuration to configure the Agent(X) so that it would divert the events/logs to a backup/Secondary collector(Z)?

**Reply by spencergilbert** [reply]
I think you'd be looking for something like sink level circuit breakers, which aren't implemented but are a feature we'd like to add: https://github.com/vectordotdev/vector/issues/8545
